src/panic.rs

```diff
@@ -1,21 +1,29 @@
 #![cfg(not(feature = "std"))]
 
-use Vec;
-use byteorder::{LittleEndian, ByteOrder};
+extern "C" {
+	fn panic(payload_ptr: *const u8, payload_len: u32) -> !;
+}
 
 /// Overrides the default panic_fmt
+#[cfg(not(feature = "panic_with_msg"))]
 #[no_mangle]
 #[lang = "panic_fmt"]
-pub extern fn panic_fmt(_fmt: ::core::fmt::Arguments, file: &'static str, line: u32, col: u32) -> ! {
-	extern "C" {
-		fn panic(payload_ptr: *const u8, payload_len: u32) -> !;
+pub extern fn panic_fmt() -> ! {
+	let msg = "Panic has occurred in Wasm module. Build Wasm with a 'panic_with_msg' feature for details";
```

Comment:
better make it as short as possible

Reply:
do we really need any constant message here? Maybe it is better to be handled on the embedder side?

Reply:
Yeah, you can just pass `1usize as *const u8` or something, and handle such pointer at runtime

Reply:
No need for this trick, nullptr should be fine as long as compiler cant see dereferences or pointer arithmetic.

Reply:
won't null pointers lead to ub?

Reply:
As I mentioned, it will lead to UB only if we try to do things like pointer arithmetic (`ptr::null().offset(x)` 😈 ) or dereferences AND compiler can see it.

So to mess things up we need not only to do mentioned above things with the pointer but also  compiler of the contract should "see" it. But contract's compiler definitely can't see through runtime boundary!